synapse/http/matrixfederationclient.py

```diff
@@ -406,7 +404,14 @@ def __init__(
         self.clock = hs.get_clock()
         self._store = hs.get_datastores().main
         self.version_string_bytes = hs.version_string.encode("ascii")
-        self.default_timeout = 60
+        self.default_timeout = hs.config.federation.client_timeout_ms / 1000
+
+        self.max_long_retry_delay = hs.config.federation.max_long_retry_delay_ms / 1000
```

Comment:
I feel like we shouldn't be doing the `/ 1000` and then doing `delay * 1000` again below.

Reply:
I can keep in milli but then I need to add 2 times `* 1000` on L738 and L742 so not sure it's a win.
Also it's consistent with `client_timeout` unit wise, and keep changes to a minimum.

Reply:
Another solution is to convert to seconds directly at the config layer.

Reply:
:shrug: Up to you, much easier to understand with units :fast_forward: